### PR TITLE
feat(import/export) : [FOR-355] add export/import event functions for Archive module

### PR DESCRIPTION
**Pull request by qperie**
## Describe your changes
Implement functions for import and export via Archive module :
- Add a SQL field named "original_form_id" containing the id of the form we copy
- Change all the "BEGIN;" and "COMMIT;" messages in transactions by a constants containing these values
- Implement the functions of import and export called by the Archive module

-> Needs to add i18n in entcore : cf https://github.com/opendigitaleducation/entcore/pull/298

I also wrote a documentation about this export/import system with the Archive module : https://confluence.support-ent.fr/pages/viewpage.action?pageId=127632788

## Issue ticket number and link
FOR-355 : https://jira.support-ent.fr/browse/FOR-355

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by JinDouang — APPROVED** [review]

File: formulaire-public/src/main/java/fr/openent/formulaire_public/controllers/FormulairePublicController.java [hunk 1]
```diff
@@ -24,7 +24,7 @@ public FormulairePublicController() {
     }
 
     @SecuredAction(ConsoleRights.ACCESS_RIGHT)
-    public void initSecuredActions(final HttpServerRequest request) {
+    public void initAccessRight(final HttpServerRequest request) {
```
Comment (sjollois) [inline formulaire-public/src/main/java/fr/openent/formulaire_public/controllers/FormulairePublicController.java:27]: Attention à bien prévenir les PO des conséquences de ce changement de nom de fonction

File: formulaire/src/main/java/fr/openent/formulaire/service/impl/FormulaireRepositoryEvents.java [hunk 2]
```diff
@@ -1,30 +1,523 @@
...
+        HashMap<String, JsonArray> fieldsToNull = new HashMap<>();
+        AtomicBoolean exported = new AtomicBoolean(false);
+
+        JsonArray groupsAndUserIds = new JsonArray();
+        groupsAndUserIds.add(userId);
+        if (groups != null) { groupsAndUserIds.addAll(groups); }
+
+        // Form query
+        Promise<JsonArray> promise = Promise.promise();
+        String formTableQuery = "SELECT f.* FROM " + FORM_TABLE + " f " +
```
Comment (sjollois) [inline formulaire/src/main/java/fr/openent/formulaire/service/impl/FormulaireRepositoryEvents.java:67]: tu ne mettrais pas la requête dans un service ?

File: formulaire/src/main/java/fr/openent/formulaire/service/impl/FormulaireRepositoryEvents.java [hunk 3]
```diff
@@ -1,30 +1,523 @@
...
+    }
+
+    /**
+     * Generate mappings between two fields from given data
+     * @param data data results of a transaction
+     * @param keyName name of the field to get to fill the map's keys
+     * @param valueName name of the field to get to fill the map's values
+     * @return A complete mapping between the two given fields
+     */
+    private Map<Integer, Integer> generateMapping(JsonArray data, String keyName, String valueName) {
```
Comment (sjollois) [inline formulaire/src/main/java/fr/openent/formulaire/service/impl/FormulaireRepositoryEvents.java:509]: J'aurais bien mis toutes les fonctions privées dans un service et les fonctions principales dans le repository events

**Review by sjollois — APPROVED** [review]

File: formulaire/src/main/java/fr/openent/formulaire/service/impl/FormulaireRepositoryEvents.java (hunk 2 above)
Comment (qperie) [reply formulaire/src/main/java/fr/openent/formulaire/service/impl/FormulaireRepositoryEvents.java:67]: Mais j'ai besoin de la query (pas son result) plus loin dans le fichier aussi :/

**Pull request merged**